google/container.go

```diff
@@ -114,17 +114,25 @@ func (c *Container) Put(name string, r io.Reader, size int64, metadata map[strin
 	}
 
 	w := obj.NewWriter(c.ctx)
+	w.ObjectAttrs.Metadata = merge(w.ObjectAttrs.Metadata, mdPrepped)
 	if _, err := io.Copy(w, r); err != nil {
 		return nil, err
 	}
-	w.Close()
-
-	attr, err := obj.Update(c.ctx, storage.ObjectAttrsToUpdate{Metadata: mdPrepped})
```

Comment:
I wonder if the errors we are seeing on some RPCs are because we are not checking `err = w.Close()`